src/Exception.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Dhii\File\Finder;
+
+/**
+ * An exception that is related to file finding.
+ *
+ * @since [*next-version*]
+ */
+class Exception extends \Exception implements ExceptionInterface
```

Comment:
I feel like this and the interface that it implements, should have their names prefixed to be more meaningful. Like `FinderException` or `FileFinderException`. It also prevents repetitive aliasing.